test/Edition/EditionOperatorFiltering.t.sol

```diff
@@ -0,0 +1,147 @@
+            // fallback to static list
+            console2.log("No network specified, using static list (Dec 2022 snapshot)");
+
+            filteredOperators =
+                abi.decode(
+                    hex"0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000111abe46ff893f3b2fdf1f759a8a8000000000000000000000000fed24ec7e22f573c2e08aef55aa6797ca2b3a051000000000000000000000000f42aa99f011a1fa7cda90e5e98b277e306bca83e000000000000000000000000b16c1342e617a5b6e4b631eb114483fdb289c0a4000000000000000000000000d42638863462d2f21bb7d4275d7637ee5d5541eb00000000000000000000000008ce97807a81896e85841d74fb7e7b065ab3ef0500000000000000000000000092de3a1511ef22abcf3526c302159882a4755b22000000000000000000000000cd80c916b1194beb48abf007d0b79a7238436d56",
+                    (address[])
+                );
+
+            MockRegistry mockRegistry = new MockRegistry();
+            vm.etch(CANONICAL_OPERATOR_FILTER_REGISTRY, address(mockRegistry).code);
+            MockRegistry(CANONICAL_OPERATOR_FILTER_REGISTRY).setFilteredOperators(filteredOperators);
+        }
+    }
+
+
+    function testOperatorFiltering() public {
+        IERC721 nftContract = IERC721(contractAddress);
+
+        // Try to get the current owner of the NFT, falling back to value set during setup on revert
+        try nftContract.ownerOf(tokenId) returns (address _owner) {
+            owner = _owner;
+        } catch (bytes memory) {
+            // Do nothing
+        }
+
+        for (uint256 i = 0; i < filteredOperators.length; i++) {
+            address operator = filteredOperators[i];
+            console2.log("Testing operator: ", operator);
+
+            // Try to set approval for the operator
+            vm.startPrank(owner);
+            try nftContract.setApprovalForAll(operator, true) {
+                // blocking approvals is not required, so continue to check transfers
+            } catch (bytes memory) {
+                // continue to test transfer methods, since marketplace approvals can be
+                // hard-coded into contracts
+            }
+
+            // also include per-token approvals as those may not be blocked
+            try nftContract.approve(operator, tokenId) {
+                // continue to check transfers
+            } catch (bytes memory) {
+                // continue to test transfer methods, since marketplace approvals can be
+                // hard-coded into contracts
+            }
+
+            vm.stopPrank();
+            // Ensure operator is not able to transfer the token
+            vm.startPrank(operator);
+            vm.expectRevert();
+            nftContract.safeTransferFrom(owner, address(1), tokenId);
+
+            vm.expectRevert();
+            nftContract.safeTransferFrom(owner, address(1), tokenId, "");
+
+            vm.expectRevert();
+            nftContract.transferFrom(owner, address(1), tokenId);
```

Comment:
why three times?

Reply:
3 different methods! We need to make sure they all implement the blocking

Reply:
ahh ok got it i didn't process the last one wasn't "safe"